panic in handler_test.go

I just got this running tests locally.  It is intermittent, I can't reproduce it.  Also, it looks like the panic is inside the test, so its not something that will happen in the field.

```
--- FAIL: TestPostNDJSONLogWarning (0.00s)
panic: interface conversion: interface {} is *api.LogPostStatus, not *api.LogPostWarning [recovered]
	panic: interface conversion: interface {} is *api.LogPostStatus, not *api.LogPostWarninggoroutine 261 [running]:
testing.tRunner.func1(0xc0001f6100)
	/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x15c4440, 0xc000341770)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/brimsec/zq/zqd_test.TestPostNDJSONLogWarning(0xc0001f6100)
	/Users/aswan/src/zq/zqd/handlers_test.go:438 +0x8e4
testing.tRunner(0xc0001f6100, 0x16af030)
	/usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
```

--- a/driver/mux.go
+++ b/driver/mux.go
@@ -74,7 +74,7 @@ func NewMuxOutput(ctx *proc.Context, parents []proc.Proc) *MuxOutput {
 }
 
 func (m *MuxOutput) Complete() bool {
-	return len(m.ctx.Warnings) == 0 && m.runners <= 0
+	return len(m.ctx.Warnings) == 0 && m.runners == 0
 }
 
 func (m *MuxOutput) N() int {

--- a/driver/mux.go
+++ b/driver/mux.go
@@ -74,7 +74,7 @@ func NewMuxOutput(ctx *proc.Context, parents []proc.Proc) *MuxOutput {
 }
 
 func (m *MuxOutput) Complete() bool {
-	return m.runners <= 0
+	return len(m.ctx.Warnings) == 0 && m.runners <= 0
 }
 
 func (m *MuxOutput) N() int {

